core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -99,12 +104,28 @@ public int read() throws IOException {
     if (mPosition == mLength) { // at end of file
       return -1;
     }
-    updateStream();
-    int result = mBlockInStream.read();
-    if (result != -1) {
-      mPosition++;
+    Set<WorkerNetAddress> lostWorkers = new HashSet<>();
+    int retryLeft = Configuration.getInt(PropertyKey.USER_BLOCK_WORKER_CLIENT_READ_RETRY);
+    boolean retryCurrentWorker = true;
+    while (true) {
```

Comment:
Use `CountingRetry`?

Reply:
+1

Reply:
we can even make the `CountingRetry` a field to share retries across reads and avoid constructing a `CountingRetry` on every read.